src/ecpool_worker_sup.erl

```diff
@@ -18,17 +18,21 @@
 
 -behaviour(supervisor).
 
--export([start_link/3]).
+-export([start_link/4]).
 
 -export([init/1]).
 
-start_link(Pool, Mod, Opts) ->
-    supervisor:start_link(?MODULE, [Pool, Mod, Opts]).
+-export([pool_size/1]).
 
-init([Pool, Mod, Opts]) ->
+start_link(Pool, Mod, Opts, InitialConnectResultReceiver) ->
+    supervisor:start_link(?MODULE, [Pool, Mod, Opts, InitialConnectResultReceiver]).
+
+init([Pool, Mod, Opts, InitialConnectResultReceiver]) ->
     WorkerSpec = fun(Id) ->
                      #{id => {worker, Id},
-                       start => {ecpool_worker, start_link, [Pool, Id, Mod, Opts]},
+                       start => {ecpool_worker,
+                                 start_link,
+                                 [Pool, Id, Mod, Opts, InitialConnectResultReceiver]},
```

Comment:
missed this before.
putting it in the child spec will contaminate the caller's mailbox if the worker had to restart after initialized.

Reply:
Good point that I did not think about before. Do you have any suggestion for an alternative solution?

Reply:
... I saw your suggestion here: https://github.com/emqx/ecpool/pull/50#discussion_r1664721629

I think this will work.